### PR TITLE
Converted into API 3, added old keys removing

### PR DESCRIPTION
**Pull request by Hojang2**

### CODE REVIEW

File: copr-gpg-download.py [hunk 1]
```diff
@@ -1,53 +1,103 @@
...
 import argparse
 from argparse import RawTextHelpFormatter
 import os.path
 
-def get_gpg(project):
-    url = be_url_tmpl.format(**{'username': project.owner, 'projectname': project.name})
-    r = requests.get(url)
+
+def get_gpg(url):
+    print(url.replace("@", "%40"))
```
Comment (xsuchy) [inline copr-gpg-download.py]: Debug leftover?

File: copr-gpg-download.py [hunk 2]
```diff
@@ -1,53 +1,103 @@
...
+
+    r = requests.get(url.replace("@", "%40"))
```
Comment (xsuchy) [inline copr-gpg-download.py]: Why is the replace needed?

File: copr-gpg-download.py [hunk 3]
```diff
@@ -1,53 +1,103 @@
...
+    else:
+        run_2()
+
+
+parser = argparse.ArgumentParser(description='Download GPG keys for COPR projects.',
+                                 formatter_class=RawTextHelpFormatter)
 parser.add_argument('-f', '--file', action='store',
                     help='store keys to a file instead of printing to stdout')
+parser.add_argument('--api', action='store', default='2',
+                    help='version of api(2/3) you want to use.')
```
Comment (xsuchy) [inline copr-gpg-download.py]: No need to look back. Just wipe away v2 and leave just v3

File: copr-gpg-download.py [hunk 4]
```diff
@@ -1,53 +1,103 @@
...
+
         _offset += _limit
 
-parser = argparse.ArgumentParser(description='Download GPG keys for COPR projects.', formatter_class=RawTextHelpFormatter)
+
+def run_3():
+    projects = cli.project_proxy.get_list(ownername=None, pagination={"limit": 10})
+    while projects:
+        if not projects:
+            break
```
Comment (xsuchy) [inline copr-gpg-download.py]: ???? These 3 lines above are not needed at all.

Comment (xsuchy) [reply copr-gpg-download.py]: Ahh, I see it is because of pagination.

File: copr-gpg-download.py [hunk 5]
```diff
@@ -1,53 +1,103 @@
...
+                not_found(file_name)
+            else:
+                gpg_out(args.isolate_files, file_name, url)
+
         _offset += _limit
 
-parser = argparse.ArgumentParser(description='Download GPG keys for COPR projects.', formatter_class=RawTextHelpFormatter)
+
+def run_3():
+    projects = cli.project_proxy.get_list(ownername=None, pagination={"limit": 10})
```
Comment (xsuchy) [inline copr-gpg-download.py]: Can we have limit=100 like in ver2?

File: copr-gpg-download.py [hunk 6]
```diff
@@ -1,51 +1,63 @@
 #!/usr/bin/python3
 
+from copr.v3 import Client
+from copr.v3.pagination import next_page
 from copr import create_client2_from_params
```
Comment (xsuchy) [inline copr-gpg-download.py]: This import is no longer used and can be removed (hint you can run pylint-3 on this file).

File: copr-gpg-download.py [hunk 7]
```diff
@@ -1,51 +1,63 @@
...
-def gpg_out(isolate_file, project):
+
+def gpg_out(isolate_file, file_name, url):
     if isolate_file:
-        file_name = "copr-{0}-{1}.gpg".format(project.owner, project.name)
         if not os.path.isfile(file_name):
-            f = open(file_name, 'w')
-            f.write(get_gpg(project))
-            f.close()
+            with open(file_name, "w") as f:
```
Comment (xsuchy) [inline copr-gpg-download.py]: I tried to run:  `pushd keys/copr && ../../copr-gpg-download.py --isolate-files && popd` and I got:
```
Traceback (most recent call last):
  File "../../copr-gpg-download.py", line 89, in <module>
    main()
  File "../../copr-gpg-download.py", line 46, in main
    gpg_out(args.isolate_files, file_name, url)
  File "../../copr-gpg-download.py", line 22, in gpg_out
    with open(file_name, "w") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'keys/copr/copr-@mock-mock.gpg'
```

File: copr-gpg-download.py [hunk 8]
```diff
@@ -58,16 +70,21 @@ def main():
                     help='only download gpg keys for projects of this name')
 parser.add_argument('--isolate-files', action='store_true',
                     help='Each GPG file is stored in separate file.')
+parser.add_argument('--path', action='store', default='keys/copr/',
```
Comment (xsuchy) [inline copr-gpg-download.py]: This is the culprit. To have the same behaviour the default path should be ./

File: copr-gpg-download.py [hunk 9]
```diff
@@ -1,51 +1,63 @@
...
+
         for project in projects:
-            gpg_out(args.isolate_files, project)
-        _offset += _limit
+            file_name = args.path + "copr-{0}-{1}.gpg".format(project.ownername, project.name)
+            url = be_url_tmpl.format(**{'username': project.ownername, 'project_name': project.name})
+            if "404 Not Found" in get_gpg(url):
+                not_found(file_name)
+            else:
+                gpg_out(args.isolate_files, file_name, url)
```
Comment (xsuchy) [inline copr-gpg-download.py]: You first download the file in the if-condition and here once again.

**Comment by Hojang2** [comment]
Ok, hope I have repaired all issues and thank you for showing pushd and popd.

**Comment by Hojang2** [comment]
Rebased and repaired #!/usr/bin/python3.

**Comment by xsuchy** [comment]
Thank you.

**Pull request merged**